database/import_test.go

```diff
@@ -0,0 +1,324 @@
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "409554438998846785912755332",
+				rowIndex:           833233,
+				numberOfBlocks:     1,
+			},
+			expectedBlock:        nil,
+			expectedErrorMessage: "strconv.ParseInt: parsing \"536870912a\": invalid syntax",
+		},
+		{
+			testID: 7,
+			name:   "wrong character in nonce, should return error",
+			data: testCaseData{
+				blockRecord: [][]string{
+					{"536870912", "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885", "3035389718a", "403300437", "1708954423"},
+				},
+				previousBlockHash:  "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "409554438998846785912755332",
+				rowIndex:           833233,
+				numberOfBlocks:     1,
+			},
+			expectedBlock:        nil,
+			expectedErrorMessage: "strconv.ParseUint: parsing \"3035389718a\": invalid syntax",
+		},
+		{
+			testID: 8,
+			name:   "wrong character in bits, should return error",
+			data: testCaseData{
+				blockRecord: [][]string{
+					{"536870912", "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885", "3035389718", "403300437a", "1708954423"},
+				},
+				previousBlockHash:  "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "409554438998846785912755332",
+				rowIndex:           833233,
+				numberOfBlocks:     1,
+			},
+			expectedBlock:        nil,
+			expectedErrorMessage: "strconv.ParseUint: parsing \"403300437a\": invalid syntax",
+		},
+		{
+			testID: 9,
+			name:   "wrong character in timestamp, should return error",
+			data: testCaseData{
+				blockRecord: [][]string{
+					{"536870912", "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885", "3035389718", "403300437", "1708954423a"},
+				},
+				previousBlockHash:  "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "409554438998846785912755332",
+				rowIndex:           833233,
+				numberOfBlocks:     1,
+			},
+			expectedBlock:        nil,
+			expectedErrorMessage: "strconv.ParseInt: parsing \"1708954423a\": invalid syntax",
+		},
+		{
+			testID: 10,
+			name:   "too long markleroot, should return error",
+			data: testCaseData{
+				blockRecord: [][]string{
+					{"536870912", "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885a", "3035389718", "403300437", "1708954423a"},
+				},
+				previousBlockHash:  "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "409554438998846785912755332",
+				rowIndex:           833233,
+				numberOfBlocks:     1,
+			},
+			expectedBlock:        nil,
+			expectedErrorMessage: "max hash string length is 64 bytes",
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
similar as above, you should change this to use `t.Run(tc.name, func(t *testing.T) {...})` so we would get the feedback which test is failing

Reply:
Done as well as above